src/lib.rs

```diff
@@ -33,48 +30,68 @@ pub struct UrlEncodedQuery;
 /// Use it like this: `req.get_ref::<UrlEncodedBody>()`
 pub struct UrlEncodedBody;
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum ErrorType{
```

Comment:
Also a different name, like `UrlDecodingError`, would be better. If you can think of something that fits better, feel free.